recipes/server.rb

```diff
@@ -34,6 +34,11 @@
 
   ssh_hosts << n['ipaddress'] if n['keys']
 
+  if(!n['fqdn'] || !n['ipaddress'] || !n['keys'])
+    Chef::Log.warn("Not setting up client keys for `#{n}` which is missing required information.")
+    next
+  end
```

Comment:
I'm not wild about this syntax. It seems really unclear and too much like "magic."

I'd rather have a helper library that checks for whatever is required at the top of the recipe and then bails out early if the requirements aren't met.